src/main/java/com/algolia/search/models/HttpRequest.java

```diff
@@ -0,0 +1,82 @@
+package com.algolia.search.models;
+
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+public class HttpRequest {
```

Comment:
Could benefit Nullability annotations (same for `HttpResponse`) :)

Reply:
Same for any model which would be manipulated by our users, when its fields are not covered by methods with Nullability annotations

Reply:
Could you please elaborate? 🤔 Do you mean to add annotations on the constructors parameters?

Reply:
I mean adding annotations wherever our users might benefit it:
- Constructor parameters of classes they could instantiate (not convinced it would be relevant here, as I don't think they have to create a `new HTTPRequest()` anytime)
- Annotations on public fields / public {g,s}etters that the users would manipulate to avoid unnecessary null-checks (e.g. if `headers` is never `null`, would mark `getHeaders()` as `@NonNull`) 

I think this can be applied more consistently in some classes, e.g. :
- `new Variant(...)` takes a `@NonNull String index` in constructor
-> `Variant.getIndex()` could be annotated as `@NonNull`
-> `Variant.setIndex()` should have its param annotated as `@NonNull`

You might want to use Intellij's `Infer Nullity` to quickly generate consistent annotations based on your current codebase :upside_down_face:

Reply:
Thanks for the clarification! Neat idea. I'll look at it.